test/e2e/specs/editor/blocks/list.spec.js

```diff
@@ -0,0 +1,872 @@
+			`<!-- wp:list -->
+<ul><li>1<ul><li>a<ul><li>i</li></ul></li></ul></li></ul>
+<!-- /wp:list -->`
+		);
+
+		await page.keyboard.press( 'Backspace' );
+		await page.keyboard.press( 'Backspace' ); // Should be at level 1.
+
+		expect( await editor.getEditedPostContent() ).toBe(
+			`<!-- wp:list -->
+<ul><li>1<ul><li>a</li><li></li></ul></li></ul>
+<!-- /wp:list -->`
+		);
+
+		await page.keyboard.press( 'Backspace' ); // Should be at level 0.
+
+		expect( await editor.getEditedPostContent() ).toBe(
+			`<!-- wp:list -->
+<ul><li>1<ul><li>a</li></ul></li><li></li></ul>
+<!-- /wp:list -->`
+		);
+
+		await page.keyboard.press( 'Backspace' ); // Should be at level 1.
+
+		expect( await editor.getEditedPostContent() ).toBe(
+			`<!-- wp:list -->
+<ul><li>1<ul><li>a</li></ul></li></ul>
+<!-- /wp:list -->`
+		);
+
+		await page.keyboard.press( 'Backspace' );
+		await page.keyboard.press( 'Backspace' ); // Should be at level 0.
+
+		expect( await editor.getEditedPostContent() ).toBe(
+			`<!-- wp:list -->
+<ul><li>1</li><li></li></ul>
+<!-- /wp:list -->`
+		);
+
+		await page.keyboard.press( 'Backspace' ); // Should be at level 0.
+
+		expect( await editor.getEditedPostContent() ).toBe(
+			`<!-- wp:list -->
+<ul><li>1</li></ul>
+<!-- /wp:list -->`
+		);
+
+		await page.keyboard.press( 'Backspace' );
+		await page.keyboard.press( 'Backspace' ); // Should remove list.
+
+		expect( await editor.getEditedPostContent() ).toBe( '' );
+
+		// That's 9 key presses to create the list, and 9 key presses to remove
+		// the list. ;)
+	} );
+
+	test( 'should place the caret in the right place with nested list', async ( {
+		editor,
+		page,
+	} ) => {
+		await page.locator( 'role=button[name="Add default block"i]' ).click();
+		await page.keyboard.type( '* 1' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( ' a' );
+		await page.keyboard.press( 'ArrowUp' );
+		await page.keyboard.press( 'Enter' );
+		// The caret should land in the second item.
+		await page.keyboard.type( '2' );
+
+		expect( await editor.getEditedPostContent() ).toBe(
+			`<!-- wp:list -->
+<ul><li>1</li><li>2<ul><li>a</li></ul></li></ul>
+<!-- /wp:list -->`
+		);
+	} );
+
+	test( 'should not indent list on space with modifier', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await page.locator( 'role=button[name="Add default block"i]' ).click();
+
+		await page.keyboard.type( '* 1' );
+		await page.keyboard.press( 'Enter' );
+		await pageUtils.pressKeyWithModifier( 'shift', 'Space' );
+
+		expect( await editor.getEditedPostContent() ).toBe(
+			`<!-- wp:list -->
+<ul><li>1</li><li> </li></ul>
+<!-- /wp:list -->`
+		);
+	} );
+
+	test( 'should only convert to list when shortcut ends with space', async ( {
+		editor,
+		page,
+	} ) => {
+		await page.locator( 'role=button[name="Add default block"i]' ).click();
+
+		// Tests the shortcut with a non breaking space.
+		await page.keyboard.type( '* ' );
+
+		/* eslint-disable no-irregular-whitespace */
```

Comment:
Why do we need to disable this?

Reply:
The space after Asterisk is U+00a0, so eslint must be disabled. @kevin940726

Reply:
Could we inline the unicode char point? e.g. `\u00a0`